rosflight/src/mavrosflight/mavrosflight.cpp

```diff
@@ -36,26 +36,43 @@
 
 #include <rosflight/mavrosflight/mavrosflight.h>
 
+#if defined(USE_ROS)
+#include <rosflight/ros_logger.h>
+#elif defined(STANDALONE)
+#include <rosflight/mavrosflight/logger_interface.h>
+#endif
+
 #include <ros/ros.h>
 
 namespace mavrosflight
 {
 using boost::asio::serial_port_base;
 
-MavROSflight::MavROSflight(MavlinkComm &mavlink_comm, uint8_t sysid /* = 1 */, uint8_t compid /* = 50 */) :
+template <typename DerivedLogger>
+MavROSflight<DerivedLogger>::MavROSflight(MavlinkComm &mavlink_comm,
+                                          LoggerInterface<DerivedLogger> &logger,
+                                          uint8_t sysid /* = 1 */,
+                                          uint8_t compid /* = 50 */) :
   comm(mavlink_comm),
-  param(&comm),
-  time(&comm),
+  param(&comm, logger),
+  time(&comm, logger),
   sysid_(sysid),
   compid_(compid)
 {
   //! \todo Fix constructors so that we can open the port in here
   // comm.open();
 }
 
-MavROSflight::~MavROSflight()
+template <typename DerivedLogger>
+MavROSflight<DerivedLogger>::~MavROSflight()
 {
   comm.close();
 }
 
+#if defined(USE_ROS)
+template class MavROSflight<rosflight::ROSLogger>;
+#elif defined(STANDALONE)
```

Comment:
What kinds of error messages do we get if we don't define `STANDALONE` or `USE_ROS`?

Would it be helpful to add?
```
#else
#error unknown logging backend supplied for mavrosflight.  Available defines include USE_ROS or STANDALONE
#endif
```

Reply:
Great point, the errors happen during linking so the messages are not helpful.

Reply:
@superjax should this only be in mavrosflight.cpp or anywhere these defines are checked?

Reply:
Try to trace to the first place that these are going to get included (basically, any public header).  Internal implementation headers are probably fine.

Reply:
Ok, I'll keep it in mavrosflight.cpp then and remove the error message from time_manager and param_manager